src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      case action.type === 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case action.type === 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+    }
+  }
+
```

Comment:
CheckList
![image](https://user-images.githubusercontent.com/80700247/201472737-e61a26a1-3e42-4e3a-b9f2-2f821f7b2ead.png)

Reply:
please be attentive in reading the comments. Review what should be in the default block